File: material/Chapter01/isPrime.c
#include <stdio.h>

int isPrime(int n)
{
  int m;
  int found_factor;

  found_factor = 0;

  for (m = 2; m <= n - 1; m++)
  {
    if (n % m == 0)
    {
      found_factor = 1;
      break;
    }
  }

  return !found_factor;
}

int main()
{
  int n;

  n = 37;

  printf("%d\n", isPrime(n));
  return 0;
}

File: material/Chapter01/isPrime.hs
-- Is n a prime number?
isPrime :: Integer -> Bool
isPrime n = factors n == []
  where
    factors :: Integer -> [Integer]
    factors n = [ m | m <- [2..n-1], mod n m == 0 ]


main :: IO ()
main = do
  let n = 43
  print (isPrime n)
